stylus/ui-app/forms.styl

```diff
@@ -5,9 +5,158 @@
  * This file contains all needs for forms, inputs, labels...
  */
 
+@import '../ui-base/icons'
+
 checkbox-size = 20px
 
-$checkbox
+$form
+    .coz-form
+        position relative
+        margin   1em 0 0.5em
+
+        // Managing input width with flex
+        display         flex
+        flex-direction  column
+
+    label
+        display        block
+        text-transform uppercase
+        color          grey-11
+        font-size      .9em
+        padding        .55em 0
+        margin-top     1em
+
+    .coz-error-label
+        display none
+        color   red
+
+    .coz-errors
+        color red
+
+    a, a:visited
+        color               dodger-blue
+        text-decoration     none
+
+
+$form-button
+    .coz-form-controls
+        width      70%
+        padding     0 15%
+        display    flex
+        margin-top 1.5em
+
+        button
+            width       100%
+            margin      0.4em
+            cursor      pointer
+
+    [role=button],
+    button
+        // mutualize behavior between button and a
+        box-sizing        border-box
+        border            0
+        display           block
+        padding           0.9em 0.75em
+        min-height        (40/basefont)em
+        border-radius     4px
+        background-color  blue
+        font-family       "Source Sans Pro", sans-serif
+        text-align        center
+        text-transform    uppercase
+        color             white
+        text-decoration   none
+
+        &.cancel
+            background-color grey-01
+            color            grey-08
+
+        &.danger
+            background-color rgba(255, 93, 70, 0.1)
+            border           solid 1px rgba(255, 93, 71, 0.1)
+            color            #ff5d47
+            margin           1em 0 3em
+            width            80%
+
+        &.flat
+            background      none
+            padding         0em
+            min-height      0
+            color           blue
+            text-transform  uppercase
+            text-decoration none
+            cursor          pointer
+
+        &[disabled]
+        &[aria-disabled=true]
+            opacity .5
+            cursor  default
+
+        &[aria-busy=true]
+            position relative
+
+            &::after
+                position     static
+                width        (16/basefont)em
+                height       (16/basefont)em
+                margin-left  (8/basefont)em
+                margin-right (-24/basefont)em
```

Comment:
Yep, I totally agree. My primary though about them is to use a shorthand that we can later extend if we need, w/o the need to refactor all code that only use margin/padding. But maybe it's totally overkill. Whatever we choose, we need to make all codebase compliant 😉